sdk/nodejs/dagger/api/api.ts

```diff
@@ -0,0 +1,309 @@
+
+        resolve(queryFlatten(response));
+      })
+    )
+
+    const result = await computeQuery;
+
+    return result
+  }
+}
+
+export default class Api extends BaseApi {
+  
+  container(args?: {id: any}): Container {
+
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'container',
+      args
+      }
+    ]
+
+    return new Container(this._queryTree)
+  }
+
+  git(args: {url: string}): Git {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'git',
+      args
+      }
+    ]
+
+    return new Git(this._queryTree)
+  }
+
+  host(): Host {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'host',
+      }
+    ]
+
+    return new Host(this._queryTree)
+  }
+
+}
+
+class Host extends BaseApi {
+  workdir(args?: {exclude?: Array<string>, include?: Array<string>}): Directory {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'workdir',
+      args
+      }
+    ]
+
+    return new Directory(this._queryTree)
+  }
+}
+
+class Git extends BaseApi {
+  branch(args: {name: string}): Tree {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+        operation: 'branch',
+        args
+      }
+    ]
+
+    return new Tree(this._queryTree)
+  }
+
+}
+class Tree extends BaseApi {
+  tree(): File {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+        operation: 'tree'
+      }
+    ]
+
+    return new File(this._queryTree)
+  }
+}
+class File extends BaseApi {
+  file(args: {path: string}): Contents {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+        operation: 'file',
+        args
+      }
+    ]
+
+    return new Contents(this._queryTree) 
+  }
+}
+
+class Container extends BaseApi {
+
+  exec(args: ContainerExecArgs): Container {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'exec',
+      args
+      }
+    ]
+    
+    return new Container(this._queryTree)
+  }
+
+  from(args: { address: String } ): Container {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'from',
+      args
+      }
+    ]
+    
+    return new Container(this._queryTree)
+  }
+
+  fs(): Directory {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'fs',
+      }
+    ]
+    
+    return new Directory(this._queryTree)
+  }
+
+  withFS(args: {id: string}): Container {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'withFS',
+      args
+      }
+    ]
+    
+    return new Container(this._queryTree)
+  }
+
+  withMountedDirectory(args: {path: string, source: string}): Container {
+    this._queryTree = [
+      ...this._queryTree,
+      {
+      operation: 'withMountedDirectory',
+      args
+      }
+    ]
+    
+    return new Container(this._queryTree)
+  }
+
+  async id(): Promise<Record<string, string>> {
```

Comment:
What about fields that do not return strings?

For instance, `id` should be a `ContainerID` instead of a `string`

Reply:
Also -- not extremely familiar with TypeScript -- but why SDK users should do `{ id } = ...id()` instead of `id = ...id()`?

I typically see that's being used for multiple returns, but since `id` will always return a single field, is that still the convention?

Reply:
> Also -- not extremely familiar with TypeScript -- but why SDK users should do { id } = ...id() instead of id = ...id()?
> 
> I typically see that's being used for multiple returns, but since id will always return a single field, is that still the convention?

In my case since the query `...id()` will return the last key value of the graphql response `{id: "myid"}`, I'm destructuring my object by getting directly the id value: 
`const { id } = ...id()`  -- id const is equal to "myIdValue"
`const id = ....id()` -- id const is equal of { id: "myIdValue" }

At the end both solution are valid.

Reply:
> What about fields that do not return strings?
> 
> For instance, `id` should be a `ContainerID` instead of a `string`

You're right on this one. I'm doing a review on types.

Reply:
> In my case since the query ...id() will return the last key value of the graphql response {id: "myid"}, I'm destructuring my object by getting directly the id value:

What I'm asking is which one makes the most sense for a node API: should `id()` return `{ id: "myIdValue" }` or should it return `"myIdValue"`?

Reply:
Following our conversation from discord (shorturl.at/noHIW), as long graphql return only one result, there is no need of returning an object. So I'll return only `"myIdValue"`